### ISSUE TITLE
imread not loading image on Xcode OpenCV 3.2

### ISSUE THREAD

**Issue by 79rpm** [opencv/opencv#9124]
##### System information (version)

- OpenCV => 3.2
- Operating System / Platform => OSX Sierra  10.12.3
- Compiler => Xcode 8.3.3

##### Detailed description

imread() does not load the jpeg image. The name of the file is passed via an argument.

##### Steps to reproduce

Run the following code, checkers.jpg is attached
![checkers](https://user-images.githubusercontent.com/7407275/27991006-8c75cf94-642e-11e7-898a-a82778436515.jpg)

// OpenCV
#include "opencv2/imgproc.hpp"
#include "opencv2/highgui.hpp"

using namespace cv;
using namespace std;

int main(int argc, const char * argv[]) {
       
    Mat image;
    image = imread(argv[1]);
    
    if(image.empty())
    {
        cout <<  "Could not open or find the image: " << argv[1] << std::endl ;
        return -1;
    }
    namedWindow("Display Loaded Image");
    imshow("Display Loaded Image", image);
    
    //Wait forever until we hit a key
    waitKey(0);
    
    return 0;
}

>>>>>>> Code returns >>>>>>>>>

Could not open or find the image: checkers.jpg
Program ended with exit code: 255

checkers.jpg is located in the exact same directory as the application binary!!

**Comment by csukuangfj** [comment]
Is it checkers.**jpeg** or checkers.**jpg** ?

**Comment by 79rpm** [comment]
I just checked and updated the post. The file is checkers.jpg

**Comment by alalek** [comment]
Please add OpenCV build information to check enabled codecs:
```
std::cout << cv::getBuildInformation() << std::endl;
```

> located in the exact same directory as the application binary

It should be located in the "current directory".
Did you try `fopen(argv[1], "rb")` call?

**Comment by 79rpm** [comment]
Hello, Thank you for your reply. I found what the problem was. 

I initially built the OpenCV 3.2 source code and installed the resulting libraries under /usr/local/lib. Under the same repository I had the libjpeg installed as well (I need to load jpeg for image processing). The sample code that I posted above was linked with that libjpeg.

I went back and installed the opencv2.framework this time and removed any link to libjpeg. The imread() returned without error.

It has been quite hard to find one single post that would explain how to install the framework to be able to build C++ application. I found some good information in the source code regarding libjpeg. There is no need to try to link a jpeg or png library as it is supported natively (good to know). 

"On Microsoft Windows\* OS and MacOSX\*, the codecs shipped with an OpenCV image (libjpeg,
    libpng, libtiff, and libjasper) are used by default. So, OpenCV can always read JPEGs, PNGs,
    and TIFFs. On MacOSX, there is also an option to use native MacOSX image readers. But beware
    that currently these native image loaders give images with different pixel values because of
    the color management embedded into MacOSX."
